pkg/api/store/cluster/cluster_store.go

```diff
@@ -825,3 +830,79 @@ func replaceWithOldSecretIfNotExists(oldData, newData map[string]interface{}, cl
 		values.PutValue(newData, oldSecret, keys...)
 	}
 }
+
+func validateIngressConfigExtraFields(data map[string]interface{}) error {
+	if err := validateIngressConfigExtraVolumes(data); err != nil {
+		return err
+	}
+	if err := validateIngressConfigExtraVolumeMounts(data); err != nil {
+		return err
+	}
+	if err := validateIngressConfigExtraEnvs(data); err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateIngressConfigExtraVolumes(data map[string]interface{}) error {
+	var volumes []v3.ExtraVolume
+	var mapVolumes []map[string]interface{}
+	extraVolumes := values.GetValueN(data, "rancherKubernetesEngineConfig", "ingress", managementv3.IngressConfigFieldExtraVolumes)
+	if extraVolumes == nil {
+		return nil
+	}
+	if err := convert.ToObj(extraVolumes, &volumes); err != nil {
+		return httperror.WrapAPIError(err, httperror.InvalidBodyContent, "ingressExtraVolumes spec conversion error")
+	}
+	for _, vol := range volumes {
+		mapForm, err := convert.EncodeToMap(vol)
+		if err != nil {
+			return err
+		}
+		mapVolumes = append(mapVolumes, mapForm)
+	}
+	values.PutValue(data, mapVolumes, "rancherKubernetesEngineConfig", "ingress", managementv3.IngressConfigFieldExtraVolumes)
```

Comment:
we could directly put `volumes` instead of `mapVolumes` 

same for L885 and L906